.github/workflows/e2e.yml

```
@@ -0,0 +1,68 @@
+name: E2E Tests
+
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+
+on:
+  push:
+    branches: [main]
+  issue_comment:
+    types: [created]
```

@kriswest we can do this in a follow-up PR but it may be worth adding `workflow_dispatch` and allowing this test suite to be executed as part of a pre-release step in the future.
```suggestion
  push:
    branches: [main]
  issue_comment:
    types: [created]
  workflow_dispatch:
```